src/NServiceBus.Core/Routing/SingleInstanceRoundRobinDistributionStrategy.cs

```diff
@@ -11,15 +10,15 @@ public class SingleInstanceRoundRobinDistributionStrategy : DistributionStrategy
         /// <summary>
         /// Selects destination instances from all known instances of a given endpoint.
         /// </summary>
-        public override IEnumerable<UnicastRoutingTarget> SelectDestination(IList<UnicastRoutingTarget> currentAllInstances)
+        public override UnicastRoutingTarget SelectDestination(UnicastRoutingTarget[] currentAllInstances)
         {
-            if (currentAllInstances.Count == 0)
+            if (currentAllInstances.Length == 0)
             {
-                yield break;
+                return null;
             }
-            var result = currentAllInstances[(int)(index % currentAllInstances.Count)];
+            var result = currentAllInstances[(int)(index % currentAllInstances.Length)];
```

Comment:
@danielmarbach raised a dedicated PR for this: https://github.com/Particular/NServiceBus/pull/4024